### PR TITLE
[AIP-133/134/135] Test LRO responses better.

### PR DESCRIPTION
**Pull request by lukesneeringer**
This adds response type checking when the protobuf response
type is an LRO.

(It also fixes a couple of bugs I found in the process. ;-))

### CODE REVIEW

File: rules/aip0134/method_rules.go [hunk 1]
```diff
@@ -55,25 +55,31 @@ var responseMessageName = &lint.MethodRule{
 	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
 		// Rule check: Establish that for methods such as `UpdateFoo`, the response
 		// message is `Foo` or `google.longrunning.Operation`.
-		got := m.GetOutputType().GetFullyQualifiedName()
-		want := []string{
-			m.GetName()[6:],
-			"google.longrunning.Operation",
+		want := m.GetName()[6:]
```
Comment (mingzhi) [inline rules/aip0134/method_rules.go]: Consider use `strings.Replace(m.GetName(), "Update", "", 1)`:
1. It avoids runtime errors about slice out of bound, if somehow we make a mistake in `isUpdateMethod`;
2. It is better documented -- I needed sometime to figure out what is `6` for.

File: rules/aip0135/method_rules.go [hunk 2]
```diff
@@ -52,27 +53,38 @@ var responseMessageName = &lint.MethodRule{
...
-		want := []string{
+		// message is `google.protobuf.Empty` or `Foo`.
+		got := m.GetOutputType().GetName()
+		if stringset.New("Empty", "Operation").Contains(got) {
+			got = m.GetOutputType().GetFullyQualifiedName()
+		}
+		want := stringset.New(
 			"google.protobuf.Empty",
-			"google.longrunning.Operation",
 			m.GetName()[6:],
```
Comment (mingzhi) [inline rules/aip0135/method_rules.go]: Consider use `strings.Replace(m.GetName(), "Update", "", 1)`.

Comment (lukesneeringer) [reply rules/aip0135/method_rules.go]: Delete, but done.

File: rules/aip0134/method_rules.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/aip0134/method_rules.go]: Good idea. Done.

**Pull request merged**